src/main/java/jenkins/plugins/logstash/configuration/ElasticSearch.java

```diff
@@ -0,0 +1,189 @@
+package jenkins.plugins.logstash.configuration;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import hudson.Extension;
+import hudson.util.FormValidation;
+import hudson.util.Secret;
+import jenkins.plugins.logstash.Messages;
+import jenkins.plugins.logstash.persistence.ElasticSearchDao;
+
+public class ElasticSearch extends LogstashIndexer<ElasticSearchDao>
+{
+  private String username;
+  private Secret password;
+  private URL url;
+
+  @DataBoundConstructor
+  public ElasticSearch()
+  {
+  }
+
+  public URL getUrl()
+  {
+    return url;
+  }
+
+
+  private URI getUri()
+  {
+    if (url != null)
+    {
+      URI uri;
+      try
+      {
+        uri = url.toURI();
+        return uri;
+      }
+      catch (URISyntaxException e)
+      {
+        return null;
+      }
+    }
+    return null;
+  }
+
+  @DataBoundSetter
+  public void setUrl(URL uri)
+  {
+    this.url = uri;
+  }
+
+  public String getUsername()
+  {
+    return username;
+  }
+
+  @DataBoundSetter
+  public void setUsername(String username)
+  {
+    this.username = username;
+  }
+
+  public String getPassword()
+  {
+    return Secret.toString(password);
+  }
+
+  @DataBoundSetter
+  public void setPassword(String password)
+  {
+    this.password = Secret.fromString(password);
+  }
+
+  @Override
+  public boolean equals(Object obj)
+  {
+    if (obj == null)
+      return false;
+    if (this == obj)
+      return true;
+    if (getClass() != obj.getClass())
+      return false;
+    ElasticSearch other = (ElasticSearch) obj;
+    if (!Secret.toString(password).equals(other.getPassword()))
+    {
+      return false;
+    }
+    if (url == null)
+    {
+      if (other.url != null)
+        return false;
+    }
+    // String comparison is not optimal but comparing the urls directly  is
+    // criticized by findbugs as being a blocking operation
+    else if (!url.toString().equals(other.url.toString()))
```

Comment:
minor: I'd suggest to have an URI field instead and do an URL -> URI conversion in setter